app/editor/src/features/content/form/ContentForm.tsx

```diff
@@ -96,10 +94,21 @@ export const ContentForm: React.FC<IContentFormProps> = ({ contentType = Content
       const model = toModel(values);
       contentResult = !content.id ? await addContent(model) : await updateContent(model);
 
-      // Now upload the file if it exists.
       if (!!values.file) {
-        const uploadResult = await upload(contentResult, values.file);
-        setContent(toForm({ ...uploadResult, tonePools: values.tonePools }));
+        // TODO: Make it possible to upload on the initial save instead of a separate request.
+        // Upload the file if one has been added.
+        const result = await upload(contentResult, values.file);
+        setContent(toForm({ ...result, tonePools: values.tonePools }));
+      } else if (
+        !originalId &&
+        !!values.fileReferences.length &&
+        !values.fileReferences[0].isUploaded
+      ) {
+        // TODO: Make it possible to upload on the initial save instead of a separate request.
+        // A file was attached but hasn't been uploaded to the API.
+        const fileReference = values.fileReferences[0];
```

Comment:
This resolves the issue where the attached clip will be copied over to the API properly.  It's not a great solution.